tools/release-checklist

```diff
@@ -0,0 +1,327 @@
+cd $TMPDIR
+(cd ocaml-$VERSION && rm -rf experimental)
+gtar -c --owner 0 --group 0 -f ocaml-$VERSION.tar ocaml-$VERSION
+time gzip -9 <ocaml-$VERSION.tar >ocaml-$VERSION.tar.gz
+time xz <ocaml-$VERSION.tar >ocaml-$VERSION.tar.xz
+```
+
+## 8: upload the archive and compute checksums
+
+```
+scp ocaml-$VERSION.tar.{xz,gz} yquem.inria.fr:$DIST
+```
+
+To update the checksum files on the remote host, we first upload the
+release environment as a script.
+
+
+```
+rm /tmp/env.sh
+echo export MAJOR=$MAJOR >> /tmp/env.sh
+echo export MINOR=$MINOR >> /tmp/env.sh
+echo export BUGFIX=$BUGFIX >> /tmp/env.sh
+echo export PLUSEXT=$PLUSEXT >> /tmp/env.sh
+
+echo export BRANCH=$BRANCH >> /tmp/env.sh
+echo export VERSION=$VERSION >> /tmp/env.sh
+
+echo export REPO=$REPO >> /tmp/env.sh
+echo export DIST=$DIST >> /tmp/env.sh
+
+# check env.sh
+cat /tmp/env.sh
+
+scp /tmp/env.sh yquem.inria.fr:/tmp
+```
+
+and then login there to update the checksums (MD5SUM, SHA512SUM)
+
+```
+ssh yquem.inria.fr
+source /tmp/env.sh
+cd $DIST
+
+cp MD5SUM MD5SUM.old
+md5sum ocaml-$VERSION.tar.{xz,gz} > new-md5s
+# check new-md5s to ensure that they look right, and then
+cat new-md5s >> MD5SUM
+# if everything worked well,
+rm MD5SUM.old
+
+# same thing for SHA512
+cp SHA512 SHA512.old
+cp SHA512SUM SHA512SUM.old
+sha512sum ocaml-$VERSION.tar.{xz,gz} > new-sha512s
+cat new-sha512s >> SHA512SUM
+rm SHA512SUM.old
+```
+
+## 9: update note files (technical documentation)
+
+```
+ssh yquem.inria.fr "mkdir -p $DIST/notes"
+scp INSTALL.adoc LICENSE README.adoc README.win32.adoc Changes yquem.inria.fr:$DIST/notes/
+```
+
+## 10: upload the reference manual
+
+You don't need to do this if the previous release had the same
+$MAJOR.$MINOR ($BRANCH) value and the exact same manual -- this is frequent if
+it was a release candidate.
+
+```
+make world.opt
+make install
+export PATH="$INSTDIR/bin:$PATH"
+cd manual
+make clean
+make
+rm -rf /tmp/release
+mkdir -p /tmp/release
+RELEASENAME="ocaml-$BRANCH-" make release
+scp /tmp/release/* yquem:$DIST/
+
+# upload manual checksums
+ssh yquem.inria.fr "cd $DIST; md5sum ocaml-$BRANCH-refman* >>MD5SUM"
+ssh yquem.inria.fr "cd $DIST; sha256sum ocaml-$BRANCH-refman* >>SHA512SUM"
+```
+
+Releasing the manual online happens on `pauillac`, reached from `yquem`:
+
+```
+ssh yquem.inria.fr
+ssh pauillac
+
+cd /net/brion/infosystems/www/caml/pub/docs
+mkdir -p ocaml-$BRANCH
+cd ocaml-$BRANCH
+wget http://caml.inria.fr/pub/distrib/ocaml-$BRANCH/ocaml-$BRANCH-refman-html.tar.gz
+tar -xzvf manual-ocaml-$BRANCH-refman-html.tar.gz # this extracts into htmlman/
+mv htmlman/* -t . # move HTML content to docs/manual-caml-$BRANCH
+rmdir htmlman
+
+cd /net/brion/infosystems/www/caml/pub/docs/manual-ocaml
+ln -sf manual-ocaml-$BRANCH manual-ocaml
+```
+
+## 11: prepare web announce for the release
+
+For production releases, you should get in touch with ocaml.org to
+organize the webpage for the new release. See
+
+  <https://github.com/ocaml/ocaml.org/issues/819>
```

Comment:
We need a bit more on the list than this, I think, because for 4.06 the web pages were not fixed for quite some time and not everything was noticed at once.

Reply:
In the issue someone volunteered to write more detailed steps, and I'm waiting for them to include here. ;-)